src/apply-networking-linux.py

```diff
@@ -215,32 +216,53 @@ def set_network_config_file(self, network_data, reset_to_dhcp=False):
         with open(self.config_file, 'w') as config_file:
             config_file.write(template_string)
 
-    def apply_network_config(self, network_data, reset_to_dhcp=False):
-        links = {}
-        for link in network_data["links"]:
-            os_link_name = get_os_net_interface_by_mac(
-                link["ethernet_mac_address"])
-            if not os_link_name:
-                raise Exception(
-                    "Link could not be found " + link["ethernet_mac_address"])
-            link["os_link_name"] = os_link_name
-            links[link["id"]] = link
-
-            LOG("Apply config for link " + os_link_name)
-            base_cmd = ["ip", "link", "set", "dev", os_link_name]
-
-            mtu_cmd = base_cmd + ["mtu", link["mtu"]]
-            out, err, exit_code = execute_process(mtu_cmd, shell=False)
+    def _get_device_for_link(self, network_data, link):
+        for n_link in network_data["links"]:
+            LOG(">> %r --> %r"  %(n_link["id"], link))
+            if n_link["id"] == link:
+                return get_os_net_interface_by_mac(n_link["ethernet_mac_address"])
+        raise Exception("could not find device for link %s" % link)
```

Comment:
Let s keep consistency with other logs / exceptions capitalizing the first letter.